simulariumio/smoldyn/smoldyn_converter.py

```diff
@@ -137,7 +134,18 @@ def _parse_objects(
 
         result.n_agents[time_index] = agent_index
         result.n_timesteps = time_index + 1
-        return result
+
+        if input_data.meta_data.scale_factor is None:
+            # If scale factor wasn't provided, calculate one
+            scale_factor = TrajectoryConverter.calculate_scale_factor(
+                result.positions, result.radii, result.n_agents, is_2D=(not is_3D)
+            )
+        else:
+            scale_factor = input_data.meta_data.scale_factor
+        result.radii = scale_factor * result.radii
+        result.positions = scale_factor * result.positions
```

Comment:
is the above 10 lines or so repeated in all the other converters? possible DRY refactor?